packages/keyring/src/pairs.ts

```diff
@@ -13,7 +13,7 @@ export default class Pairs implements KeyringPairs {
   readonly #map: KeyringPairMap = {};
 
   public add (pair: KeyringPair): KeyringPair {
-    this.#map[pair.publicKey.toString()] = pair;
+    this.#map[decodeAddress(pair.address).toString()] = pair;
```

Comment:
We want to lookup on the underlying `publicKey` bytes - this allows addresses for any encoding to be retrieved. So you can add ssFormat 42 addresses, change to 1 and still retrieve and have it available.